cmake/FindWin32SslRuntime.cmake

```diff
@@ -1,66 +1,47 @@
     file(TO_CMAKE_PATH "$ENV{PROGRAMFILES}" _programfiles)
     set(_OPENSSL_ROOT_PATHS
-      "C:/Tools/vcpkg/installed/x64-windows/bin"
-      "${_programfiles}/OpenSSL-Win64"
-      "C:/OpenSSL-Win64/"
-      )
+            "$ENV{VCPKG_PACKAGES_DIR}/x64-windows/bin"
+            "C:/OpenSSL-Win64/bin"
+            "C:/OpenSSL-Win64"
+            "C:/Tools/vcpkg/installed/x64-windows/bin"
+            "${_programfiles}/OpenSSL-Win64"
+            )
     unset(_programfiles)
-  else( CMAKE_SIZEOF_VOID_P EQUAL 8 )
-    # target win32
-    set(_OPENSSL_ROOT_HINTS
-      ${OPENSSL_ROOT_DIR}
-      "[HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\OpenSSL (32-bit)_is1;Inno Setup: App Path]"
-      ENV OPENSSL_ROOT_DIR
-      )
+  elseif("${CMAKE_GENERATOR_PLATFORM}" STREQUAL "Win32")
+    message(STATUS "Looking for OpenSSL for ${CMAKE_GENERATOR_PLATFORM}")
+    set(_OPENSSL_ROOT_HINTS ${OPENSSL_ROOT_DIR} ENV OPENSSL_ROOT_DIR)
     file(TO_CMAKE_PATH "$ENV{PROGRAMFILES}" _programfiles)
     set(_OPENSSL_ROOT_PATHS
-      "C:/Tools/vcpkg/installed/x86-windows/bin"
-      "${_programfiles}/OpenSSL"
-      "${_programfiles}/OpenSSL-Win32"
-      "C:/OpenSSL/"
-      "C:/OpenSSL-Win32/"
-      )
+            "$ENV{VCPKG_PACKAGES_DIR}/x86-windows/bin"
+            "C:/OpenSSL-Win32/bin"
+            "C:/OpenSSL-Win32"
+            "C:/OpenSSL"
+            "C:/Tools/vcpkg/installed/x86-windows/bin"
+            "${_programfiles}/OpenSSL"
+            "${_programfiles}/OpenSSL-Win32"
+            )
     unset(_programfiles)
-  endif( CMAKE_SIZEOF_VOID_P EQUAL 8 )
-
+  else()
+    message(WARNING "Unknown OS_PATH: $ENV{OS_PATH}, Can't find OpenSSL correctly")
+  endif()
 else ()
-  set(_OPENSSL_ROOT_HINTS
-    ${OPENSSL_ROOT_DIR}
-    ENV OPENSSL_ROOT_DIR
-    )
-endif ()
-
-set(_OPENSSL_ROOT_HINTS_AND_PATHS
-    HINTS ${_OPENSSL_ROOT_HINTS}
-    PATHS ${_OPENSSL_ROOT_PATHS}
-    )
-
-# For OpenSSL < 1.1, they are named libeay32 and ssleay32 and even if the dll is 64bit, it's still suffixed as *32.dll
-# For OpenSSL >= 1.1, they are named libcrypto and libssl with no suffix
-if( CMAKE_SIZEOF_VOID_P EQUAL 8 )
-  # target win64
-  FIND_FILE(WIN32SSLRUNTIME_LIBEAY NAMES libcrypto-1_1-x64.dll libcrypto.dll libeay32.dll ${_OPENSSL_ROOT_HINTS_AND_PATHS})
-  FIND_FILE(WIN32SSLRUNTIME_SSLEAY NAMES libssl-1_1-x64.dll libssl.dll ssleay32.dll ${_OPENSSL_ROOT_HINTS_AND_PATHS})
-else( CMAKE_SIZEOF_VOID_P EQUAL 8 )
-  # target win32
-  FIND_FILE(WIN32SSLRUNTIME_LIBEAY NAMES libcrypto-1_1.dll libcrypto.dll libeay32.dll ${_OPENSSL_ROOT_HINTS_AND_PATHS})
-  FIND_FILE(WIN32SSLRUNTIME_SSLEAY NAMES libssl-1_1.dll libssl.dll ssleay32.dll ${_OPENSSL_ROOT_HINTS_AND_PATHS})
-endif( CMAKE_SIZEOF_VOID_P EQUAL 8 )
+  set(_OPENSSL_ROOT_HINTS ${OPENSSL_ROOT_DIR} ENV OPENSSL_ROOT_DIR)
+endif()
+
+message(STATUS "Looking for OpenSSL @ ${CMAKE_GENERATOR_PLATFORM} in ${_OPENSSL_ROOT_PATHS}")
+if("${CMAKE_GENERATOR_PLATFORM}" STREQUAL "x64")
```

Comment:
this part seems to be outside of the WIN32 check but doesn't seem platform agnostic at all, isn't this entire file only used on windows anyway?

Reply:
I've changed it to ensure we're in a win set, but yeah it's only ever included on windows